tests/Tests/ORM/Functional/SecondLevelCacheStrictTest.php

```diff
@@ -0,0 +1,141 @@
+    {
+        $metadata = $this->_em->getClassMetaData(Country::class);
+
+        if ($cacheUsage === 0) {
+            $metadataCacheReflection = new ReflectionProperty(ClassMetadataInfo::class, 'cache');
+            $metadataCacheReflection->setAccessible(true);
+            $metadataCacheReflection->setValue($metadata, null);
+        } else {
+            if ($cacheUsage === ClassMetadata::CACHE_USAGE_READ_WRITE) {
+                $this->tmpDir = sys_get_temp_dir() . DIRECTORY_SEPARATOR . self::class;
+                $this->secondLevelCacheFactory->setFileLockRegionDirectory($this->tmpDir);
+            }
+
+            $metadata->enableCache(['usage' => $cacheUsage]);
+        }
+    }
+
+    /** @param 'INSERT'|'UPDATE'|'DELETE' $type */
+    private function assertQueryCountByType(string $type, int $expectedCount): void
+    {
+        $queries = array_filter($this->getQueryLog()->queries, static function (array $entry) use ($type): bool {
+            return strpos($entry['sql'], $type) === 0;
+        });
+
+        self::assertCount($expectedCount, $queries);
+    }
+
+    /**
+    /* @param SupportedCacheUsage $cacheUsage
+     *
+     * @dataProvider cacheUsageProvider
+     */
+    public function testInsert(int $cacheUsage): void
+    {
+        $this->setupCountryModel($cacheUsage);
+
+        self::assertQueryCountByType('INSERT', 0);
+
+        $this->loadFixturesCountries();
+
+        self::assertCount(2, $this->countries);
+        self::assertQueryCountByType('INSERT', 2);
+    }
+
+    /**
+    /* @param SupportedCacheUsage $cacheUsage
+     *
+     * @dataProvider cacheUsageProvider
+     */
+    public function testDelete(int $cacheUsage): void
+    {
+        $this->setupCountryModel($cacheUsage);
+        $this->loadFixturesCountries();
+
+        $c1 = $this->_em->find(Country::class, $this->countries[0]->getId());
+        $c2 = $this->_em->find(Country::class, $this->countries[1]->getId());
+
+        $this->_em->remove($c1);
+        $this->_em->remove($c2);
+        $this->_em->flush();
+
+        self::assertQueryCountByType('DELETE', 2);
+    }
+
+    /**
+    /* @param SupportedCacheUsage $cacheUsage
+     *
+     * @dataProvider cacheUsageProvider
+     */
+    public function testUpdate(int $cacheUsage): void
+    {
+        $this->setupCountryModel($cacheUsage);
+        $this->loadFixturesCountries();
+
+        $c1 = $this->_em->find(Country::class, $this->countries[0]->getId());
+        $c2 = $this->_em->find(Country::class, $this->countries[1]->getId());
+
+        $c1->setName('Czech Republic');
+        $c2->setName('Hungary');
+
+        $this->_em->persist($c1);
+        $this->_em->persist($c2);
+        $this->_em->flush();
+
+        self::assertQueryCountByType('UPDATE', 2);
+    }
+
+    public static function cacheUsageProvider(): array
+    {
+        return [
+            [0],
+            [ClassMetadata::CACHE_USAGE_NONSTRICT_READ_WRITE],
```

Comment:
The title of this test contains `Strict`, but here you are testing `NONSTRICT`…

Reply:
I test all 3 - no cache at all, non-strict and strict cache. What name would you recommed?

Reply:
Yeah why not